command/plugin.go

```diff
@@ -21,19 +21,21 @@ func (c *PluginCommand) Help() string {
 Usage: vault plugin <subcommand> [options] [args]
 
   This command groups subcommands for interacting with Vault's plugins and the
-  plugin catalog. Here are a few examples of the plugin commands:
+  plugin catalog. The plugin catalog is divided into three types: "auth", 
+  "database", and "secret" plugins. A type must be specified on each call. Here 
+  are a few examples of the plugin commands.
 
-  List all available plugins in the catalog:
+  List all available plugins in the catalog of a particular type:
 
-      $ vault plugin list
+      $ vault plugin list database
 
-  Register a new plugin to the catalog:
+  Register a new plugin to the catalog as a particular type:
 
-      $ vault plugin register -sha256=d3f0a8b... my-custom-plugin
+      $ vault plugin register -sha256=d3f0a8b... my-custom-plugin auth
 
-  Get information about a plugin in the catalog:
+  Get information about a plugin in the catalog listed under a particular type:
 
-      $ vault plugin info my-custom-plugin
+      $ vault plugin info my-custom-plugin auth
```

Comment:
I can change it. I'm not sure what you mean by breaking compatibility though. Can you give an example of the command you're thinking of?

I like this one:
```
vault auth plugin info my-custom-plugin
```

Reply:
What about `vault plugin info auth my-plugin`? I think it's nice to keep "vault plugin info" together to indicate that this in an info query command more than a vault auth command.

Reply:
Adding to my comment above, we currently have `vault plugin list TYPE`, so it might make sense to do `plugin info TYPE NAME` as well to keep it consistent.

Reply:
Pushed in https://github.com/hashicorp/vault/pull/5536/commits/5ace05aafca1e7d60a9649c653e3a33f5d7e71bc